Tests/ApodiniTests/RequestTests/CodableTests.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  CodableTests.swift
+//  
+//
+//  Created by Moritz Schüll on 23.12.20.
+//
+
+import XCTest
+import NIO
+@testable import Apodini
+
+final class CodableTests: ApodiniTests, EncodableContainerVisitor {
+    struct ActionHandler: Handler {
+        var message: String
+
+        func handle() -> Action<String> {
+            .final(message)
+        }
+    }
+
+    struct FutureBasedHandler: Handler {
+        var eventLoop: EventLoop
+        var message: String
+
+        func handle() -> Action<EventLoopFuture<EventLoopFuture<EventLoopFuture<String>>>> {
+            // this tests if the `EventLoopFutureUnwrapper` properly unwraps multiple nested EventLoopFutures
+            // I hope no one would ever do that, but its possible, thus we need to handle it properly
+            .send(
```

Comment:
I would consider this the other way around and to only allow that an `Action` carries an `Encodable` that is only a concrete type and not an `EventLoopFuture`. Looking over this example I am not 100% convinced that moving all the responsibility to `Encodable` is the perfect solution as It won't allow us to express this constraint on compile time as you have already mentioned in the PR description. 
Even if we replace `Encodable` with a custom solution I would consider this problem and the `Encodable` problem at two different layers. 

But to be honest I am not sure how to perfectly solve this at this point. I will give it a though in the morning as I have already experimented a bit with it while taking a look at the PR and will let you know if I have a solution that would be feasible in the near future to not block any other PRs/features relying on this.

Back to the `EventLoopFuture` nesting: I would consider `EventLoopFuture` the elements that can be nested on the outside and `Action`s can only contain `Encodable` types that are not `Action`s or `EventLoopFuture`s. 
This would also solve part of the `EventLoopFuture` problem mentioned in #97 as we could unwrap all `EventLoopFuture`s and allow `ResponseTransformers` to either only work on the `Encodable` type or `Action` where the `Element` is just an `Encodable` type and not an `Action` or `EventLoopFuture`.

If you e.g. consider a database access: The type of action is not known at the outer layer as it might depend on the elements retrieved from the database. If there are multiple elements it might be a `.send` while it could also be a `.finish` when there is only one element in the database and the `Handler` is trying to model a service side stream. In general the type and content of `Action` might be reliant on information only available in `EventLoopFutures` and therefore I would consider the possible signature the other way around: `EventLoopFuture<EventLoopFuture<EventLoopFuture<Action< String >>>>`